src/coreComponents/physicsSolvers/multiphysics/MultiphasePoroelasticSolver.cpp

```diff
@@ -0,0 +1,308 @@
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "Coupling method. Valid options:\n* " + EnumStrings< CouplingTypeOption >::concat( "\n* " ) );
+
+  m_linearSolverParameters.get().mgr.strategy = "MultiphasePoroelastic";
+  m_linearSolverParameters.get().mgr.separateComponents = true;
+  m_linearSolverParameters.get().mgr.displacementFieldName = keys::TotalDisplacement;
+  m_linearSolverParameters.get().dofsPerNode = 3;
+}
+
+void MultiphasePoroelasticSolver::registerDataOnMesh( Group & meshBodies )
+{
+  GEOSX_UNUSED_VAR( meshBodies );
+}
+
+void MultiphasePoroelasticSolver::setupDofs( DomainPartition const & domain,
+                                             DofManager & dofManager ) const
+{
+  GEOSX_MARK_FUNCTION;
+  m_solidSolver->setupDofs( domain, dofManager );
+  m_flowSolver->setupDofs( domain, dofManager );
+
+  dofManager.addCoupling( keys::TotalDisplacement,
+                          CompositionalMultiphaseBase::viewKeyStruct::elemDofFieldString(),
+                          DofManager::Connector::Elem );
+}
+
+void MultiphasePoroelasticSolver::setupSystem( DomainPartition & domain,
+                                               DofManager & dofManager,
+                                               CRSMatrix< real64, globalIndex > & localMatrix,
+                                               array1d< real64 > & localRhs,
+                                               array1d< real64 > & localSolution,
+                                               bool const setSparsity )
+{
+//  if( m_precond )
+//  {
+//    m_precond->clear();
+//  }
+
+  // setup monolithic coupled system
+  SolverBase::setupSystem( domain, dofManager, localMatrix, localRhs, localSolution, setSparsity );
+
+//  if( !m_precond && m_linearSolverParameters.get().solverType != LinearSolverParameters::SolverType::direct )
+//  {
+//    createPreconditioner();
+//  }
+}
+
+void MultiphasePoroelasticSolver::implicitStepSetup( real64 const & time_n,
+                                                     real64 const & dt,
+                                                     DomainPartition & domain )
+{
+  m_flowSolver->implicitStepSetup( time_n, dt, domain );
+  m_solidSolver->implicitStepSetup( time_n, dt, domain );
+
+}
+
+void MultiphasePoroelasticSolver::implicitStepComplete( real64 const & time_n,
+                                                        real64 const & dt,
+                                                        DomainPartition & domain )
+{
+  m_solidSolver->implicitStepComplete( time_n, dt, domain );
+  m_flowSolver->implicitStepComplete( time_n, dt, domain );
+}
+
+void MultiphasePoroelasticSolver::postProcessInput()
+{
+  SolverBase::postProcessInput();
+
+  m_flowSolver = &this->getParent().getGroup< CompositionalMultiphaseBase >( m_flowSolverName );
+  m_solidSolver = &this->getParent().getGroup< SolidMechanicsLagrangianFEM >( m_solidSolverName );
+
+  m_solidSolver->setEffectiveStress( 1 );
```

Comment:
if we keep a flag in both the mechanics and the flow solvers ( it seems to me that we can move some stuff to the coupled solver and remove those flags) I would make sure that the flags are the same (`m_poroElastic` or whatever we prefer). Obviously, if we decide to remove the flags and call the coupled kernels directly in the poroelastic solver then it makes no sense changing them.

Reply:
The only reason for keeping the flag `m_poroelasticFlag` is the split operator approach based on the fixed stress algorithm for the single phase coupled solver. If we agree on removing it (and disabling the corresponding integrated test), we could remove even now.